src/vm/exceptionhandling.cpp

```diff
@@ -4663,6 +4664,11 @@ VOID DECLSPEC_NORETURN DispatchManagedException(PAL_SEHException& ex)
                     // Unwind to the managed caller of the helper
                     PAL_VirtualUnwind(&frameContext, NULL);
                 }
+#ifdef _TARGET_ARM_
```

Comment:
That's what I meant. Currently, we do the same unwind / adjust context in both HandleHardwareException - to get the context for the FaultingExceptionFrame and the DispatchManagedException - to get the first managed frame. We can do it just once, but we need to let DispatchManagedException know it is called from the HardwareExceptionHandler, since the flag in the CONTEXT indicating we are in the hardware exception context at that moment is no longer set. I'll send a PR for the change today and you can then rebase your change on top of that.